### ISSUE TITLE
Course Management API

### ISSUE THREAD

**Issue by arshadpatel** [arshadpatel/codeX100#15]
Create a set of APIs to manage courses. This includes adding new courses, updating course details, and retrieving available courses for users.

**Tasks:**

- Create CRUD endpoints for courses (Create, Read, Update, Delete).
- Ensure proper validation for course data (e.g., course name, price).

**Comment by arshadpatel** [comment]
Hii @geekyharsh05 You can jump in and start working on the issue directly. Feel free to start working on the issue, and I look forward to seeing your contributions!

**Comment by geekyharsh05** [comment]
@arshadpatel Sure

**Comment by prateek168** [comment]
@geekyharsh05 Can i take this issue as i am beginning with my OSS journey i will be grateful if i can do this?

**Comment by geekyharsh05** [comment]
@prateek168 Yeah is that, So Congratulation on the starting of your OSS Journey Brother, Yeah you can take it. @arshadpatel Can you please assign it to him.
